error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/PrimitiveRules.java

```diff
@@ -556,4 +553,49 @@ String after(long i, int radix) {
       return Long.toUnsignedString(i, radix);
     }
   }
+
+  /** Prefer JDK's {@link Arrays#compareUnsigned(byte[], byte[])} over third-party alternatives. */
+  // XXX: This rule will yield non-compilable code if the result of the replaced expression is
+  // dereferenced. Investigate how to make this safe.
```

Comment:
IIRC I have a branch somewhere with some experiments. For these specific cases I estimate the risk to be "moderate".